students/Ajay_Randhawa/lesson09/donor_models.py

```diff
@@ -0,0 +1,91 @@
+class Donor():
+    '''
+    this class parses information for a single donor from the donorCollection class
+    it also creates the donor objects
+    '''
+    def __init__(self, name, donation):
+        self.name = name
+        self.donation_count = 1
+        self.total_donation = donation
+        global x
+        if x != 1:
+            self.send_thankyou(name, donation)
+        self.avg_donation = self.average_donation(donation, self.donation_count)
+
+    def send_thankyou(self, donor, donation):
+        return print(f'\nThank you {donor} for your generous donation of {donation}\n')
+    
+    def average_donation(self, donation, count):
+        average_donation = round(donation / count)
+        return average_donation
+
+    def add_donation(self, donor, donation):
+        self.donation_count = self.donation_count + 1
+        self.total_donation = self.total_donation + donation
+        self.avg_donation = self.average_donation(self.total_donation, self.donation_count)
+        self.send_thankyou(donor, donation)
+
+
+
+class DonorCollection(): 
+    """
+    Stores the data for all donors. At initilizaiton, takes a dict and parses the data 
+    in the class.
+    """
+    def __init__(self, dict1):
+        self.donorlist = {}
+        self.instance = []
+        self.donorlist = dict1
+        print(dict1)
+        global x
+        for donor, donation in self.donorlist.items():
+            # self.instance_name = donor.replace(" ", "_")
+            x = 1
+            self.instance_name = Donor(donor, donation[0])
+            self.instance.append(self.instance_name)
```

Comment:
should not need to have class attribute self.instance_name

you should store donor objects only, `self.donors.append(Donor(donor, donation[0]))`